assets/js/custom/ProgramCredits.js

```diff
@@ -1,50 +1,207 @@
 import $ from 'jquery'
+import Swal from 'sweetalert2'
+import { showSnackbar } from '../components/snackbar'
 /* global Routing */
 
-export function ProgramCredits (programId) {
+export function ProgramCredits (programId, translationSaved, translationDeleted, closeEditorPrompt,
+  btnDiscardText, cancelText, keepText, saveOnLanguageChangeText, creditsSelect) {
   const credits = $('#credits')
   const editCreditsUI = $('#edit-credits-ui')
   const editCredits = $('#edit-credits')
+  const editCreditsButton = $('#edit-credits-button')
+  const closeCreditsEditorButton = $('#close-credits-editor-button')
+  const editCreditsSubmitButton = $('#edit-credits-submit-button')
   const editCreditsError = $('#edit-credits-error')
   const descriptionCreditsContainer = $('#description-credits-container')
   const showMoreToggle = $('#descriptionShowMoreToggle')
   const descriptionHeadline = $('#description-headline')
 
-  $('.edit-credits-button').on('click', () => {
-    if (editCreditsUI.hasClass('d-none')) {
-      descriptionCreditsContainer.hide()
-      descriptionHeadline.hide()
-      editCreditsUI.removeClass('d-none')
-      showMoreToggle.addClass('d-none')
-    } else {
-      descriptionCreditsContainer.show()
-      descriptionHeadline.show()
-      editCreditsUI.addClass('d-none')
-      handleShowMore()
-    }
+  let previousCreditsIndex = 0
+  let languages = {}
+
+  initCreditsSelect()
+
+  function initCreditsSelect () {
+    creditsSelect.listen('MDCSelect:change', () => {
+      if (!$('#edit-credits').is(':visible') || creditsSelect.selectedIndex === previousCreditsIndex) {
+        return
+      }
+
+      if ($('#edit-credits').val().trim() === $('#credits').text().trim()) {
+        previousCreditsIndex = creditsSelect.selectedIndex
+        setNewCredits()
+      } else {
+        Swal.fire({
+          title: saveOnLanguageChangeText,
+          icon: 'question',
+          showDenyButton: true,
+          showCloseButton: true,
+          allowOutsideClick: true,
+          backdrop: true,
+          confirmButtonText: keepText,
+          denyButtonText: btnDiscardText
+        }).then((result) => {
+          if (result.isConfirmed) {
+            previousCreditsIndex = creditsSelect.selectedIndex
+          } else if (result.isDenied) {
+            previousCreditsIndex = creditsSelect.selectedIndex
+            setNewCredits()
+          } else if (result.isDismissed) {
+            creditsSelect.selectedIndex = previousCreditsIndex
+          }
+        })
+      }
+    })
+  }
+
+  $(document).ready(function () {
+    getLanguages()
   })
 
-  $('#edit-credits-submit-button').on('click', () => {
+  function getLanguages () {
+    $.ajax({
+      url: '../languages',
+      type: 'get',
+      success: function (data) {
+        languages = data
+        populateSelector()
+      }
+    })
+  }
+
+  function setNewCredits () {
+    $.ajax({
```

Comment:
Can we disable the input field or have some indication that the data is loading while the ajax call is in progress?

Reply:
Good idea, I disabled the text field until the ajax call is complete.